examples/tokens/ERC1155ownable.vy

```diff
@@ -0,0 +1,358 @@
+    operator: address = msg.sender
+    
+    for i in range(BATCH_SIZE):
+        if i >= len(ids):
+            break
+        self.balances[ids[i]][receiver] = amounts[i]
+    
+    log TransferBatch(operator, ZERO_ADDRESS, receiver, ids, amounts)
+
+## burn ##
+@external
+def burn(id: uint256, amount: uint256):
+    # @dev burn one or more token with a certain ID
+    # @dev the amount of tokens will be deducted from the holder's balance
+    # @param receiver the account that will receive the minted token
+    # @param id the ID of the token to burn
+    # @param amount of tokens to burnfor this ID
+    assert not self.paused, "The contract has been paused"
+    assert self.balances[id][msg.sender] > 0 , "caller does not own this ID"
+    operator: address = msg.sender
+    fromBalance: uint256 = self.balances[id][msg.sender]
+    assert fromBalance >= amount, "ERC1155: burn amount exceeds balance"
+    self.balances[id][msg.sender] = fromBalance - amount
+
+    log TransferSingle(operator, msg.sender, ZERO_ADDRESS, id, amount)
+    
+@external
+def burnBatch(ids: DynArray[uint256, BATCH_SIZE], amounts: DynArray[uint256, BATCH_SIZE]):
+    # @dev burn a batch of tokens with the passed IDs
+    # @dev this can be burning non fungible tokens or reducing the balance of existing non-fungible token IDs in the contract
+    # @dev inside the loop ownership will be checked for each token. We can not burn tokens we do not own
+    # @param ids array of ids for the tokens to burn
+    # @param amounts array of amounts of tokens for each ID in the ids array
+    assert not self.paused, "The contract has been paused"
+    assert len(ids) == len(amounts), "ERC1155: ids and amounts length mismatch"
+    operator: address = msg.sender 
+    
+    for i in range(BATCH_SIZE):
+        if i >= len(ids):
+            break
+        assert self.balances[ids[i]][msg.sender] > 0 , "Caller does not own this ID or ZERO balance"
+        id: uint256 = ids[i]
+        amount: uint256 = amounts[i]
+        fromBalance: uint256 = self.balances[id][msg.sender]
+        assert fromBalance >= amount, "ERC1155: burn amount exceeds balance"
+        self.balances[id][msg.sender] = fromBalance - amount
+    
+    log TransferBatch(operator, msg.sender, ZERO_ADDRESS, ids, amounts)
+
+## approval ##
+@external
+def setApprovalForAll(owner: address, operator: address, approved: bool):
+    # @dev set an operator for a certain NFT owner address
+    # @param account the NFT owner address
+    # @param operator the operator address
+    assert owner == msg.sender, "You can only set operators for your own account"
+    assert not self.paused, "The contract has been paused"
+    assert owner != operator, "ERC1155: setting approval status for self"
+    self.operatorApprovals[owner][operator] = approved
+    log ApprovalForAll(owner, operator, approved)
+
+@external 
+@view
+def isApprovedForAll(account: address, operator: address) -> bool:
```

Comment:
One thing that you could do is change the `operatorApprovals` public variable to `isApprovedForAll`, and then you don't even need this function because of the public getter